op-service/txmgr/txmgr.go

```diff
@@ -210,56 +260,109 @@ func (m *SimpleTxManager) send(ctx context.Context, candidate TxCandidate) (*typ
 		m.metr.RPCError()
 		return nil, fmt.Errorf("failed to get gas price info: %w", err)
 	}
-	gasFeeCap := CalcGasFeeCap(basefee, gasTipCap)
-
-	// TODO: If we apply the accessList manually, it's hard to predict and react to other issues,
-	// such as gas prices, due to subsequent code modifications.
-	// To avoid this, we need to add logic to calculate the accessList automatically using `eth_createAccessList`.
-	rawTx := &types.DynamicFeeTx{
-		ChainID:    m.chainID,
-		To:         candidate.To,
-		GasTipCap:  gasTipCap,
-		GasFeeCap:  gasFeeCap,
-		Data:       candidate.TxData,
-		Value:      candidate.Value,
-		AccessList: candidate.AccessList,
-	}
+	gasFeeCap := CalcGasFeeCap(baseFee, gasTipCap)
 
-	m.l.Info("Creating tx", "to", rawTx.To, "from", m.Config.From)
+	gasLimit := candidate.GasLimit
 
 	// If the gas limit is set, we can use that as the gas
-	if candidate.GasLimit != 0 {
-		rawTx.Gas = candidate.GasLimit
-	} else {
+	if gasLimit == 0 {
 		// Calculate the intrinsic gas for the transaction
 		gas, err := m.backend.EstimateGas(ctx, ethereum.CallMsg{
 			From:      m.Config.From,
 			To:        candidate.To,
-			GasFeeCap: gasFeeCap,
 			GasTipCap: gasTipCap,
-			Data:      rawTx.Data,
-			Value:     rawTx.Value,
+			GasFeeCap: gasFeeCap,
+			Data:      candidate.TxData,
+			Value:     candidate.Value,
 		})
 		if err != nil {
 			return nil, fmt.Errorf("failed to estimate gas: %w", err)
 		}
-		rawTx.Gas = gas
+		gasLimit = gas
 	}
 
-	return m.signWithNextNonce(ctx, rawTx)
+	var sidecar *types.BlobTxSidecar
+	var blobHashes []common.Hash
+	if len(candidate.Blobs) > 0 {
+		if candidate.To == nil {
+			return nil, errors.New("blob txs cannot deploy contracts")
+		}
+		if sidecar, blobHashes, err = MakeSidecar(candidate.Blobs); err != nil {
+			return nil, fmt.Errorf("failed to make sidecar: %w", err)
+		}
+	}
+
+	var txMessage types.TxData
+	if sidecar != nil {
+		if blobBaseFee == nil {
+			return nil, fmt.Errorf("expected non-nil blobBaseFee")
+		}
+		blobFeeCap := calcBlobFeeCap(blobBaseFee)
+		message := &types.BlobTx{
+			To:         *candidate.To,
+			Data:       candidate.TxData,
+			Gas:        gasLimit,
+			BlobHashes: blobHashes,
+			Sidecar:    sidecar,
+		}
+		if err := finishBlobTx(message, m.chainID, gasTipCap, gasFeeCap, blobFeeCap, candidate.Value); err != nil {
+			return nil, fmt.Errorf("failed to create blob transaction: %w", err)
+		}
+		txMessage = message
+	} else {
+		// TODO: If we apply the accessList manually, it's hard to predict and react to other issues,
+		// such as gas prices, due to subsequent code modifications.
+		// To avoid this, we need to add logic to calculate the accessList automatically using `eth_createAccessList`.
+		txMessage = &types.DynamicFeeTx{
+			ChainID:    m.chainID,
+			To:         candidate.To,
+			GasTipCap:  gasTipCap,
+			GasFeeCap:  gasFeeCap,
+			Value:      candidate.Value,
+			Data:       candidate.TxData,
+			AccessList: candidate.AccessList,
+			Gas:        gasLimit,
+		}
```

Comment:
```suggestion
		txMessage = &types.DynamicFeeTx{
			ChainID:   m.chainID,
			To:        candidate.To,
			GasTipCap: gasTipCap,
			GasFeeCap: gasFeeCap,
			Value:     candidate.Value,
			Data:      candidate.TxData,
			Gas:       gasLimit,
			// [Kroma: START]
			// TODO: If we apply the accessList manually, it's hard to predict and react to other issues,
			// such as gas prices, due to subsequent code modifications.
			// To avoid this, we need to add logic to calculate the accessList automatically using `eth_createAccessList`.
			AccessList: candidate.AccessList,
			// [Kroma: END]
		}
```